optimizely/client/client.go

```diff
@@ -133,6 +121,82 @@ func (o *OptimizelyClient) GetEnabledFeatures(userContext entities.UserContext)
 	return enabledFeatures, nil
 }
 
+// GetFeatureVariableString returns string feature variable value
+func (o *OptimizelyClient) GetFeatureVariableString(featureKey string, variableKey string, userContext entities.UserContext) (value string, err error) {
+	value, valueType, err := o.getFeatureVariable(featureKey, variableKey, userContext)
+	if err != nil {
+		return "", err
+	}
+	if valueType != "string" {
+		return "", fmt.Errorf("Variable value for key %s is wrong type", variableKey)
+	}
+	return value, err
+}
+
+func (o *OptimizelyClient) getFeatureVariable(featureKey string, variableKey string, userContext entities.UserContext) (value string, valueType string, err error) {
+
+	defer func() {
+		if r := recover(); r != nil {
+			errorMessage := fmt.Sprintf(`Optimizely SDK is panicking with the error "%s"`, string(debug.Stack()))
+			err = errors.New(errorMessage)
+			logger.Error(errorMessage, err)
+		}
+	}()
+
+	if isValid, err := o.isValidClient("getFeatureVariable"); !isValid {
+		return "", "", err
+	}
+
+	projectConfig := o.configManager.GetConfig()
+
+	feature, err := projectConfig.GetFeatureByKey(featureKey)
+	if err != nil {
+		logger.Error("Error retrieving feature", err)
+		return "", "", err
+	}
+
+	variable, err := projectConfig.GetVariableByKey(featureKey, variableKey)
+	if err != nil {
+		logger.Error("Error retrieving variable", err)
+		return "", "", err
+	}
+
+	var featureValue = variable.DefaultValue
+
+	featureDecisionContext := decision.FeatureDecisionContext{
+		Feature:       &feature,
+		ProjectConfig: projectConfig,
+	}
+
+	featureDecision, err := o.decisionService.GetFeatureDecision(featureDecisionContext, userContext)
+	if err == nil {
+		if v, ok := featureDecision.Variation.Variables[variable.ID]; ok && featureDecision.Variation.FeatureEnabled {
+			featureValue = v.Value
+		}
+	}
+
+	if featureValue == "" {
```

Comment:
Is this still needed? Empty strings should be OK. I think we should revert the above flow where if `err != nil` then we return this error message

Reply:
done! i'l do that for other getFeatureVariable API's, getFeatureVariableString doesn't require type-casting so i don't think we need it.